lizmap/modules/lizmap/lib/Project/QgisProject.php

```diff
@@ -1464,25 +1464,43 @@ protected function getFieldConfiguration($layerXml)
         return $edittypes;
     }
 
+    protected function getTabFromAttributes($attributes)
+    {
+        $tab = array();
+
+        foreach ($attributes as $key => $value) {
+            $tab[(string) $key] = (string) $value;
+        }
+
+        return $tab;
+    }
+
     protected function getEditType($layerXml)
     {
-        $edittypes = $layerXml->edittypes;
+        $edittypes = $layerXml->xpath('.//edittypes');
+        $edittypes = $edittypes[0];
         $editTab = array();
 
-        foreach ($edittypes as $edittype) {
+        foreach ($edittypes->edittype as $edittype) {
             $attributes = $edittype->attributes();
-            $editTab[$attributes->name] = array();
-            $editTab[$attributes->name]['eddittype'] = $edittype;
-            // New QGIS 2.4 edittypes : use widgetv2type property
-            if (property_exists($edittype->attributes(), 'widgetv2type')) {
-                $editTab[$attributes->name]['widgetv2configAttr'] = $edittype->widgetv2config->attributes();
-                $editTab[$attributes->name]['fieldEditType'] = (string) $edittype->attributes()->widgetv2type;
+            $name = (string) $attributes->name;
+            $type = (string) $attributes->widgetv2type;
+            $editTab[$name] = array();
+            // foreach ($edittype as $key => $value) {
+            //     $editTab[$name]['edittype'][$key] = (object)$this->getTabFromAttributes($edittype->$key->attributes());
+            // }
+            $editTab[$name]['edittype'] = (object) $this->getTabFromAttributes($edittype->attributes());
+            $editTab[$name]['edittype']->widgetv2config = (object) $this->getTabFromAttributes($edittype->widgetv2config->attributes());
```

Comment:
why duplicate this content? Since you put the same value into widgetv2configAttr below, you should remove this line

Reply:
It's because they are two different properties in qgisFormControl and it's easier to loop on the Json object with each property being a different element.